src/command-handlers/devices.ts

```diff
@@ -0,0 +1,91 @@
+import { connectAndPrepare } from '../database';
+import { deactivateDevices, listDevices, ListDevicesOutput } from '../endpoints';
+import { reset } from '../middleware';
+import { askConfirmReset } from '../utils';
+
+type OutputDevice = ListDevicesOutput['devices'][number] & {
+    isCurrentDevice: boolean;
+};
+
+export async function listAllDevices(options: { json: boolean }) {
+    const { secrets, deviceConfiguration } = await connectAndPrepare({ autoSync: false });
+    if (!deviceConfiguration) {
```

Comment:
i don't think you need this, the connectAndPrepare will register the device if it doesn't exist

Reply:
We need to fix the typings of `connectAndPrepare` then because right now it returns a nullable

Reply:
ah ok my bad, then leave it, i will take a look